Subject: Recognize logcheck files as the logcheck filetype
Author: James Vega <jamessan@debian.org>
Bug-Debian: http://bugs.debian.org/382986
Forwarded: no

Index: b/runtime/filetype.vim
===================================================================
--- a/runtime/filetype.vim
+++ b/runtime/filetype.vim
@@ -987,6 +987,9 @@
 " LiteStep RC files
 au BufNewFile,BufRead */LiteStep/*/*.rc		setf litestep
 
+" Logcheck files
+au BufNewFile,BufRead /etc/logcheck/*.d/*	setf logcheck
+
 " Login access
 au BufNewFile,BufRead /etc/login.access		setf loginaccess
 
Index: b/runtime/ftplugin/logcheck.vim
===================================================================
--- /dev/null
+++ b/runtime/ftplugin/logcheck.vim
@@ -0,0 +1,15 @@
+" Vim filetype plugin file
+" Language:    Logcheck
+" Maintainer:  Debian Vim Maintainers <pkg-vim-maintainers@lists.alioth.debian.org>
+" Last Change: 2008-08-30
+" License:     GNU GPL, version 2.0
+" URL:         http://git.debian.org/?p=pkg-vim/vim.git;a=blob_plain;f=runtime/ftplugin/logcheck.vim;hb=debian
+
+if exists("b:did_ftplugin")
+    finish
+endif
+
+let b:did_ftplugin = 1
+
+" Do not hard-wrap lines since logcheck requires one line per regex
+setlocal textwidth=0
