Subject: Recognize the BlockFold synid for Perl's syntax-based indenting
Forwarded: 2008-07-16
Author: Gregory Hainaut <gregory.hainaut@gmail.com>
Bug-Debian: http://bugs.debian.org/473660

Index: b/runtime/indent/perl.vim
===================================================================
--- a/runtime/indent/perl.vim
+++ b/runtime/indent/perl.vim
@@ -133,7 +133,7 @@
 	    \ || synid == "perlMatchStartEnd"
 	    \ || synid == "perlHereDoc"
 	    \ || synid =~ "^perlFiledescStatement"
-	    \ || synid =~ '^perl\(Sub\|BEGINEND\|If\)Fold'
+	    \ || synid =~ '^perl\(Sub\|BEGINEND\|Block\|If\)Fold'
 	let brace = strpart(line, bracepos, 1)
 	if brace == '(' || brace == '{'
 	  let ind = ind + &sw
@@ -148,7 +148,7 @@
       let synid = synIDattr(synID(v:lnum, bracepos, 0), "name")
       if synid == ""
 	    \ || synid == "perlMatchStartEnd"
-	    \ || synid =~ '^perl\(Sub\|BEGINEND\|If\)Fold'
+	    \ || synid =~ '^perl\(Sub\|BEGINEND\|Block\|If\)Fold'
 	let ind = ind - &sw
       endif
     endif
